### PR TITLE
Adding old tm2z styling content

### PR DESCRIPTION
**Pull request by rclark**
Hey @brandon-rhodes. This PR brings the files that define the styling that we currently have in place on the site. Looking at this has jogged my memory...

The way the tech works today is that the `project.xml` file here is the styling definition for the geologic and the ancillary data. All the data sources are vector tilesets: one of the geology, and then 2 mapbox-provided layers for terrain and streets.

What this reminded me of is that I HAD made a vector tileset of the geology and uploaded it to Mapbox, but I made that vector tileset using a piece of Mapbox software called [Studio Classic](https://docs.mapbox.com/help/glossary/mapbox-studio-classic/) that's no longer supported. This let me control all aspects of the vector tile generation process, and perform that process on my laptop before uploading the generated tiles to Mapbox servers. 

What does all this mean? Well, the XML file here might help you think about mapnik-based styling efforts, and it reiterates that I should take a closer look at [the newer Mapbox tiling service (in beta)](https://docs.mapbox.com/api/maps/#tilesets). If that service can produce a less simplified set of vector tiles, then it would be much easier to improve on the source data here in this repo and use it to drive updates to the site.

The other side of that is that would mean needing to convert the styling language from the `project.xml` Mapnik definitions to [the new style specification](https://docs.mapbox.com/mapbox-gl-js/style-spec/).

### CODE REVIEW

File: grandcanyon.tm2/project.xml [hunk 1]
```diff
@@ -0,0 +1,3238 @@
...
+
+<Parameters>
+  <Parameter name="attribution"><![CDATA[<a href="http://geomaps.wr.usgs.gov/arizona/">George Billingsley et al.</a> <a href="https://www.mapbox.com/about/maps/">© Mapbox © OpenStreetMap</a> and Ryan Clark, 2014]]></Parameter>
+  <Parameter name="bounds">-180,-85.0511,180,85.0511</Parameter>
+  <Parameter name="center">-111.9843,36.1498,11</Parameter>
+  <Parameter name="format">png8:m=h</Parameter>
+  <Parameter name="maxzoom">14</Parameter>
+  <Parameter name="minzoom">10</Parameter>
+  <Parameter name="name"><![CDATA[Geologic Map of the Grand Canyon]]></Parameter>
+  <Parameter name="source"><![CDATA[mapbox:///rclark.u33j714i,mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v5]]></Parameter>
```
Comment (rclark) [inline grandcanyon.tm2/project.xml:13]: @brandon-rhodes This is where source data is pulled, and the rest of the document is rules for styling that source data.

```
mapbox:///rclark.u33j714i,mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v5
```

This is a set of 3 vector tile datasets. The URI here is an instruction to a Mapbox rendering service to pull data from the 3 sources, blend them into a single set of vector data, then apply the styling rules from the rest of the document.

- rclark.u33j714i: This is the geologic tileset that I produced using old software
- mapbox.mapbox-terrain-v2: This is an old version of hillshade data that's provided by Mapbox
- mapbox.mapbox-streets-v5: This is a very old version of street/trail/landuse data that's provided by Mapbox

**Comment by brandon-rhodes** [comment]
Wow, thanks for all these details! My own temptation will probably be to experiment with local rendering, as it feels more future-proof to create a script that anyone can run standalone on their own machine, that's not tied to any specific service remaining up and running. But I'll see how far I get when I can next sit down with this new information.

**Comment by rclark** [comment]
Wow I just realized that all this stuff is actually already in this repo: 

- data/grandcanyon.tm2: This is the set of Studio Classic files that generates the Mapnik XML that's in this branch.

- data/grandcanyon.tm2source/data.yml: This is a set of instructions for asking Studio Classic to build the vector tileset. Notice that I actually had loaded the geojson into postgis in order to do this. I've changed laptops since then, that database isn't around anymore.

**Comment by rclark** [comment]
The results using the beta Mapbox service look pretty good.

image

image

There's some more styling work that I need to do, especially for the linework. And to use this on the site is going to require me to overhaul the site JavaScript (based on very old mapbox.js). On the plus side, I gotta say that the GL rendering that modern Mapbox client libraries uses looks _very nice_ as you zoom and pan.

I might also invest some time in a CI system that could make it easier for someone like you to make a PR in this repository to change some unit designation (which I think was the original ask), and once merged, drive the update to the map.

> to experiment with local rendering

All that said: more power to you on this! It is a fun thing to get your head wrapped around. I'll be excited to hear more about what you can do.

**Comment by brandon-rhodes** [comment]
> All that said: more power to you on this! It is a fun thing to get your head wrapped around. I'll be excited to hear more about what you can do.

On the one hand, it might not be useful for me to do too much more rendering if it duplicates work that you're able to do with the new service. On the other hand, I'd finish out the process of learning to build a tiled map, and there would now be a way for folks who don't use MapBox to do rendering of their own, and to make custom maps.

Adding a CI system would be a nice step!

**Comment by rclark** [comment]
[Here's a preview link if you'd like to see how far I got styling today](https://api.mapbox.com/styles/v1/rclark/ck97hq2xy1n941ipdlyw7x38h.html?fresh=true&title=view&access_token=pk.eyJ1IjoicmNsYXJrIiwiYSI6ImNqczlmdjFjMTAwdTQ0NG5xdnJscmx4M3MifQ.bqKE_Z9PdxKS9hv4AAT6hQ).

**Comment by brandon-rhodes** [comment]
It's pretty slow on my laptop, but I really like that you added labels, I don't remember them before. Nice! It's already recognizably on the way to being the previous map.

Unrelated: do you know the status of the geographic data itself ­— public domain, since it's USGS, or was the work done under more specific auspices (like a university)? If I do wind up rendering a map I didn't know if there were restrictions on its use or publication.

**Comment by rclark** [comment]
The source geologic data is from USGS publications, [listed on the readme](https://github.com/rclark/grand-canyon-geology/blob/gh-pages/README.md). I did some work to normalize those data into the geojson that's included in this repository. I did that work mostly on my own, to a limited extent with help from my former co-workers at the AZ State Geological Survey.

Public data, I would just be respectful to cite those original USGS publications.

**Comment by brandon-rhodes** [comment]
Oh, wow! I had just followed the main USGS link and my eyes glazed over when I saw "quadrangles", but, these aren't normal quadrangles. Now I know what my evening reading is going to be. Thanks for drawing my attention to them. And I'll treat the data as public, then, but be careful to credit the original publications and, when I use the JSON, your own work in transforming the data.

**Comment by rclark** [comment]
Closing this PR, modernization work continues in #17

**Pull request closed without merging**